lib/interfaces/uncaught.js

```diff
@@ -54,6 +54,7 @@ function handlerSetup(client, config) {
 
     errorHandlerRouter(err, em);
     client.sendError(em, handleProcessExit);
+    setTimeout(handleProcessExit, 2000);
```

Comment:
@matthewloring I added two types of test for this case:
1) A unit test file to cover API surface
2) A spawn-able test fixture which runs tests against the lifecycle of this uncaught exception behavior, including: grabbing the exception data, reporting it to the (mocked) endpoint and terminating the fork in under 2500ms.